### PR TITLE
Add support for the list failed organization invitations feature

### LINKED ISSUE #1812: OrganizationsService.ListFailedOrgInvitations
[/orgs/{org}/failed_invitations](https://docs.github.com/en/rest/reference/orgs#list-failed-organization-invitations) has the same API as `/orgs/{org}/invitations`, although we should likely expand the `Invitation` model to include failure fields. My userland copypasta "implementation:"

```go
// listFailedOrgInvitations returns a list of pending invitations.
// Copied and edited version of ListPendingOrgInvitations.
//
// GitHub API docs: https://docs.github.com/en/rest/reference/orgs#list-failed-organization-invitations
func listFailedOrgInvitations(ctx context.Context, client *Client, org string, opts *gh.ListOptions) ([]*gh.Invitation, *gh.Response, error) {
	u := fmt.Sprintf("orgs/%v/failed_invitations", org)
	u, err := addOptions(u, opts)
	if err != nil {
		return nil, nil, err
	}

	req, err := client.NewRequest("GET", u, nil)
	if err != nil {
		return nil, nil, err
	}

	var failedInvitations []*gh.Invitation
	resp, err := client.Do(ctx, req, &failedInvitations)
	if err != nil {
		return nil, resp, err
	}
	return failedInvitations, resp, nil
}
```

### PR DESCRIPTION
**Pull request by itsmatinx**
Resolves #1812

### CODE REVIEW

File: github/orgs_members.go
Comment (gmlewis) [inline github/orgs_members.go:368]:
```suggestion
// GitHub API docs: https://docs.github.com/en/rest/reference/orgs#list-failed-organization-invitations
```

File: github/orgs_members.go
Comment (gmlewis) [inline github/orgs_members.go:366]:
```suggestion
// ListFailedOrgInvitations returns a list of failed inviatations.
```

File: github/orgs_members_test.go
Comment (gmlewis) [inline github/orgs_members_test.go:748]:
```suggestion
func TestOrganizationsService_ListFailedOrgInvitations(t *testing.T) {
```

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @itsmatinx !
Just a few minor tweaks, please, and also please run `go generate ./...` and push the changes (please do not _force_ push) to this PR (as described in our `CONTRIBUTING.md` doc).

**Comment by itsmatinx** [comment]
> Thank you, @itsmatinx !
> Just a few minor tweaks, please, and also please run `go generate ./...` and push the changes (please do not _force_ push) to this PR (as described in our `CONTRIBUTING.md` doc).

Thank you for reviewing the PR. I read the `CONTRIBUTING.md` previously, but I misunderstood the generated files. I thought it would be enough to run `go generate ./...` and simply make sure everything works.

The requested changes have also been resolved.

**Review by gmlewis — APPROVED** [review]
Thank you, @itsmatinx !
LGTM.

Awaiting second LGTM before merging.

**Review by wesleimp — APPROVED** [review]
💙

**Comment by gmlewis** [comment]
Thank you, @wesleimp !
Merging.

**Pull request merged**
